### PR TITLE
(#93) new directories structure

### PR DESCRIPTION
**Pull request by KeterSCP**
Part of #93

Handles DownloadedResourceDir, ConfigDirPath, and ScreenshotsDir

We should probably add linux + MacOS to CI matrix?

### CODE REVIEW

File: Source/Tests/Paths/Setup/PathsFixture.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        yield return new object[] { CodeImp.Bloodmasters.Paths.TempDir };
+        yield return new object[] { CodeImp.Bloodmasters.Paths.DownloadedResourceDir };
+        yield return new object[] { CodeImp.Bloodmasters.Paths.ConfigDirPath };
+        yield return new object[] { CodeImp.Bloodmasters.Paths.ScreenshotsDir };
+    }
+
+    public void Dispose()
+    {
+        if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("CI")))
```
Comment (KeterSCP) [inline Source/Tests/Paths/Setup/PathsFixture.cs:15]: Is it ok?

**Comment by ForNeVeR** [comment]
> We should probably add linux + MacOS to CI matrix?

Opened #97 on that.

File: Source/Tests/Paths/Setup/PathsFixture.cs (hunk 1 above)
Comment (ForNeVeR) [reply Source/Tests/Paths/Setup/PathsFixture.cs:15]: Why, though? I think that it's okay to clean this up on CI as well.

File: Source/Shared/Paths.cs [hunk 2]
```diff
@@ -78,23 +78,33 @@ private static string GetAppBaseDir()
     public static readonly string BundledResourceDir = ContentDirPath;
 
     /// <summary>Directory for the downloaded resources. Read + write access.</summary>
-    // TODO[#93]: Better download path, like the user Downloads folder or AppData
-    public static readonly string DownloadedResourceDir = ContentDirPath;
+    public static readonly string DownloadedResourceDir = Directory.CreateDirectory(
+        Path.Combine(
+            Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+            "Bloodmasters"
+        )).FullName;
```
Comment (ForNeVeR) [inline Source/Shared/Paths.cs:85]: Let's add `Downloads` subfolder there? I don't want everything to stay in the same folder.

File: Source/Shared/Paths.cs [hunk 3]
```diff
@@ -78,23 +78,33 @@ private static string GetAppBaseDir()
...
     /// <summary>Directory with the game configuration files. Read + write access.</summary>
     public static readonly string ConfigDirPath =
         IsDevModeBuild
             ? Path.Combine(SolutionRootPath!, "Source", "Config", "Debug")
-            : Path.Combine(AppBaseDir); // TODO[#93]: Better path for r+w files in release, like AppData
+            : Directory.CreateDirectory(
+                Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
+                    "Bloodmasters"
+                )).FullName;
```
Comment (ForNeVeR) [inline Source/Shared/Paths.cs:95]: Let's add `Config` subfolder there?

File: Source/Tests/Paths/Setup/PathsFixture.cs (hunk 1 above)
Comment (KeterSCP) [reply Source/Tests/Paths/Setup/PathsFixture.cs:15]: It does clean on CI, and does not on local test runs (so local development files stay untouched)

File: Source/Tests/Paths/ConfigDirPathTests.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    [Fact(DisplayName = "ConfigDirPath should be named 'Debug' in dev mode and 'Bloodmasters' otherwise")]
+    public void ConfigDirPathShouldBeNamedDebugInDevModeAndBloodmastersOtherwise()
+    {
+        // Arrange
+        var dirName = Path.GetFileName(CodeImp.Bloodmasters.Paths.ConfigDirPath);
+
+        // Assert
+        Assert.Equal(
+            CodeImp.Bloodmasters.Paths.IsDevModeBuild ? "Debug" : "Bloodmasters",
+            dirName);
```
Comment (ForNeVeR) [inline Source/Tests/Paths/ConfigDirPathTests.cs:15]: _(Note)_ Created #98 about future of tests in dev/nondev mode.

File: Source/Tests/Paths/Setup/PathsFixture.cs (hunk 1 above)
Comment (ForNeVeR) [reply Source/Tests/Paths/Setup/PathsFixture.cs:15]: Actually you know what? I don't think we should clean these up at all.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
I have practically no concerns about this code, but let's figure out what to do about the folder cleanup. Also, some minor comments.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**